spec/app_spec.rb

```diff
@@ -24,6 +24,15 @@
         app.run
         expect(output.string).to include('X', 'O', '1', '│', '└')
       end
+      it 'prints user messages and error messages' do
+        input = StringIO.new("1\n2\n4\n5\n20\n7\n")
+        output = StringIO.new
+        ui = Ui.new(input, output)
+        app = App.new(ui)
+        app.run
+        expect(output.string).to include('move')
```

Comment:
Maybe a few line breaks in the tests, to split the test in to it's different sections (Act - Arrange - Assert), would be easier to see what is being tested at a glance.